### PR TITLE
Fix missing bypassed chain in Error

### PR DESCRIPTION
**Pull request by yshui**
Issue: https://issues.dlang.org/show_bug.cgi?id=16968
Forum discussion: https://forum.dlang.org/post/o2n7f8$2p1t$1@digitalmars.com

### CODE REVIEW

File: src/rt/dwarfeh.d [hunk 1]
```diff
@@ -373,25 +373,34 @@ extern (C) _Unwind_Reason_Code __dmd_personality_v0(int ver, _Unwind_Action acti
     if (exceptionClass == dmdExceptionClass)
     {
         auto eh = ExceptionHeader.toExceptionHeader(exceptionObject);
+        auto current_lsd = language_specific_data;
```
Comment (andralex) [inline src/rt/dwarfeh.d]: use `currentLsd` please

File: src/rt/dwarfeh.d [hunk 2]
```diff
@@ -418,6 +427,14 @@ extern (C) _Unwind_Reason_Code __dmd_personality_v0(int ver, _Unwind_Action acti
             //printf("delete %p\n", ehn);
             _Unwind_DeleteException(&ehn.exception_object); // discard ehn
         }
+        if (bypassed)
+        {
+            eh = ExceptionHeader.toExceptionHeader(exceptionObject);
+
```
Comment (andralex) [inline src/rt/dwarfeh.d]: why the whitespace?

File: test/exceptions/src/chain.d [hunk 3]
```diff
@@ -0,0 +1,64 @@
+// Author: Ali Çehreli
+// See https://forum.dlang.org/post/o2n7f8$2p1t$1@digitalmars.com
+
+import core.stdc.stdio;
+
+class TestException : Exception {
```
Comment (andralex) [inline test/exceptions/src/chain.d]: follow brace-on-its-line convention please

**Comment by yshui** [comment]
Hmm... I think I still don't understand why the check ```language_specific_data == ehn.languageSpecificData``` is needed

**Comment by andralex** [comment]
I was worried about the naming convention but then I noticed a different convention is used here.

**Comment by andralex** [comment]
Auto-merge toggled on

**Pull request merged**